[BUG] Scala compilation fails: BloomFilter.create() deprecated method treated as fatal error in GpuBloomFilterAggregate

**Describe the bug**
Build: rapids_databricks_nightly-dev-github/2161

The Maven build of `rapids-4-spark-sql` fails across all Databricks runtime variants (12.2, 13.3, 14.3, 17.3) with a Scala compilation error. `GpuBloomFilterAggregate.scala` line 116 calls `BloomFilter.create()` which is marked as deprecated. The Scala compiler treats this deprecation warning as a fatal error (`-Xfatal-warnings` or equivalent), causing `one error found` and Maven BUILD FAILURE for all shim targets (spark332db, spark341db, spark350db143, spark400db173).

Error logs:
```
[ERROR] [Error] /home/ubuntu/spark-rapids/sql-plugin/src/main/spark330/scala/org/apache/spark/sql/rapids/aggregate/GpuBloomFilterAggregate.scala:116: method create in class BloomFilter is deprecated
[ERROR] one error found
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.9.2:compile (scala-compile-first) on project rapids-4-spark-sql_2.12: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.9.2:compile failed: org.apache.commons.exec.ExecuteException: Process exited with an error: 255 (Exit value: 255)
```

**Environment details**
- Affected runtimes: Databricks 12.2, 13.3, 14.3, 17.3
- Build commit: 3c39cf5cfb6c2e629cffc96da6585a6067eafb60
- Maven: 3.6.3, Java: 17, Scala: 2.12 / 2.13
- File: sql-plugin/src/main/spark330/scala/org/apache/spark/sql/rapids/aggregate/GpuBloomFilterAggregate.scala:116

--- a/sql-plugin/src/main/spark330/scala/org/apache/spark/sql/rapids/aggregate/GpuBloomFilterAggregate.scala
+++ b/sql-plugin/src/main/spark330/scala/org/apache/spark/sql/rapids/aggregate/GpuBloomFilterAggregate.scala
@@ -112,7 +112,10 @@ object GpuBloomFilterAggregate {
 
 case class GpuBloomFilterUpdate(numHashes: Int, numBits: Long) extends CudfAggregate {
   override val reductionAggregate: ColumnVector => Scalar = (col: ColumnVector) => {
-    closeOnExcept(BloomFilter.create(numHashes, numBits)) { bloomFilter =>
+    // TODO: Address this properly in https://github.com/NVIDIA/spark-rapids/pull/14406.
+    // For now, only the v1 version of bloom-filters is supported.
+    closeOnExcept(BloomFilter.create(
+      BloomFilter.VERSION_1, numHashes, numBits, BloomFilter.DEFAULT_SEED)) { bloomFilter =>
       BloomFilter.put(bloomFilter, col)
       bloomFilter
     }

